tests/unit/general/test_umpire_mem.cpp

```diff
@@ -103,17 +108,16 @@ static void test_umpire_device_memory()
    printf("perm=%ld, temp=%ld\n", alloc_size(permanent), alloc_size(temporary));
 
    // allocates in permanent device memory
-   Vector dev_perm(num_elems, MemoryClass::DEVICE);
+   Vector dev_perm(num_elems);
+   dev_perm.Write(); // make sure device memory is allocated
```

Comment:
The idea of this example was to allocate on the device only and my understanding is that now we're allocating on the host then also on the device, is that correct? Should it just be `Vector dev_perm(num_elems, MemoryType::DEVICE_UMPIRE)`?

Reply:
The class `Memory` always needs a host pointer (at least in the current implementation) even when you explicitly ask for a specific device memory type. If you want to do pure device allocations, we'll need to think what's the best way to achieve that. However, on most systems host allocations are "allocate on use" (i.e. real memory is used only if you touch it) so always doing the host allocation is probably not a substantial overhead.